services/wfm/src/main/java/org/openkilda/wfm/topology/ping/bolt/GroupCollector.java

```diff
@@ -80,7 +89,9 @@ private void collect(Tuple input) throws AbstractException {
     private CollectorDescriptor saveCurrentRecord(Tuple input) throws AbstractException {
         final PingContext pingContext = pullPingContext(input);
 
-        long expireAt = System.currentTimeMillis() + expireDelay;
+        // expiring is only a memory leakage protection in this place
+        // all timeout responses must be fired in upstream
```

Comment:
It's better to add a comment with explanation of what this sum of timeouts and delays means.

Reply:
Good point, updated